### PR TITLE
#22 Added configuration file

### PR DESCRIPTION
**Pull request by mastrogiovanni**
In sample app endpoints and configurations are taken from there. config.yaml must be in the dist directory in order to work.
docker-compose will replace local configuration with the configuration.example.yaml file in root directory.

### CODE REVIEW

**Comment by bfabio** [comment]
@mastrogiovanni thanks! Reviewing ASAP

File: src/__tests__/index.test.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ const samlConfig: SamlConfig = {
   // decryptionPvk: fs.readFileSync("./certs/key.pem", "utf-8"),
   identifierFormat: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
   issuer: "https://spid.agid.gov.it/cd",
-  logoutCallbackUrl: "http://localhost:3000/slo",
+  // logoutCallbackUrl: "http://localhost:3000/slo",
```
Comment (bfabio) [inline src/__tests__/index.test.ts:110]: Is this on purpose?

File: src/configuration.yaml.example [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    - mobilePhone
+  
+  endpoints:
+    acs: /acs
+    error: /error
+    login: /login
+    metadata: /metadata
+    logout: /logout
+  
+
```
Comment (bfabio) [inline src/configuration.yaml.example:30]: We don't need this file anymore.

**Review by bfabio — CHANGES_REQUESTED** [review]

File: src/example.ts [hunk 3]
```diff
@@ -89,23 +84,26 @@ const samlConfig: SamlConfig = {
...
-  callbackUrl: "http://localhost:3000" + appConfig.assertionConsumerServicePath,
+  authnContext: process.env.AUTH_N_CONTEXT,
+  callbackUrl: `${process.env.ORG_URL}${process.env.ENDPOINT_ACS}`,
   // decryptionPvk: fs.readFileSync("./certs/key.pem", "utf-8"),
   identifierFormat: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
-  issuer: "https://spid.agid.gov.it/cd",
-  logoutCallbackUrl: "http://localhost:3000/slo",
-  privateCert: fs.readFileSync("./certs/key.pem", "utf-8"),
+  issuer: process.env.ORG_ISSUER,
+  // logoutCallbackUrl: conf.organization.url + "/slo",
```
Comment (bfabio) [inline src/example.ts:92]: Is this on purpose?

Comment (mastrogiovanni) [reply src/example.ts:92]: This is a suggestion that was useful for me: it is needed to cover cases when the host that will receive the SAML responses is different from the URL used for the Issuer.

File: src/__tests__/index.test.ts (hunk 1 above)
Comment (mastrogiovanni) [reply src/__tests__/index.test.ts:110]: It is a cleanup action: I noticed that variable *logoutCallbackUrl* is never used in the software and was commented so I thought that needed to be removed.

File: src/configuration.yaml.example (hunk 2 above)
Comment (mastrogiovanni) [reply src/configuration.yaml.example:30]: Yes, I'll remove it

File: src/__tests__/index.test.ts (hunk 1 above)
Comment (bfabio) [reply src/__tests__/index.test.ts:110]: I think this is used in passport-saml. I'd leave this in for now.

File: src/example.ts (hunk 3 above)
Comment (bfabio) [reply src/example.ts:92]: @mastrogiovanni can you also uncomment this line please?

**Review by mastrogiovanni — COMMENTED** [review]
Ok, and I'll use same env variable

File: src/example.ts [hunk 4]
```diff
@@ -89,23 +84,26 @@ const samlConfig: SamlConfig = {
...
-  callbackUrl: "http://localhost:3000" + appConfig.assertionConsumerServicePath,
+  authnContext: process.env.AUTH_N_CONTEXT,
+  callbackUrl: `${process.env.ORG_URL}${process.env.ENDPOINT_ACS}`,
   // decryptionPvk: fs.readFileSync("./certs/key.pem", "utf-8"),
   identifierFormat: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
-  issuer: "https://spid.agid.gov.it/cd",
-  logoutCallbackUrl: "http://localhost:3000/slo",
-  privateCert: fs.readFileSync("./certs/key.pem", "utf-8"),
+  issuer: process.env.ORG_ISSUER,
+  logoutCallbackUrl: c`${process.env.ORG_URL}/slo`,
```
Comment (bfabio) [inline src/example.ts:92]: That "c" shouldn't be there :)

**Review by bfabio — CHANGES_REQUESTED** [review]

File: src/example.ts (hunk 4 above)
Comment (mastrogiovanni) [reply src/example.ts]: Fixed

**Comment by bfabio** [comment]
Thanks @mastrogiovanni!

**Pull request merged**